taca/cli.py

```diff
@@ -0,0 +1,32 @@
+import os
+
+import click
+import taca.log
+
+from pkg_resources import iter_entry_points
+
+from taca.utils import config
+
+
+@click.group()
+# Priority for the configuration file is: environment variable > -c option > default
+@click.option('-c', '--config-file',
+			  default=os.path.join(os.environ['HOME'], '.taca/taca.yaml'),
+			  envvar='TACA_CONFIG',
+			  type=click.File('r'),
+			  help='Path to TACA configuration file')
+@click.pass_context
+def cli(ctx, config_file):
+	""" Tool for the Automation of Storage and Analyses """
+
+	ctx.obj = {}
+	ctx.obj['config'] = config.load_yaml_config(config_file)
+	log_file = ctx.obj['config'].get('log', {}).get('log_file', None)
+	if log_file:
+		level = ctx.obj['config'].get('log').get('log_level', 'INFO')
+		taca.log.init_logger_file(log_file, level)
+
+
+#Add subcommands dynamically to the CLI
+for entry_point in iter_entry_points('taca.subcommands'):
```

Comment:
I got ahead of you, hehe! This is the best practise solution (as far as I know) to implement a plugin/entry points loader:

```python
class EntryPointsCLI(click.MultiCommand):

  """Add subcommands dynamically to a CLI via entry points."""

  def list_commands(self, ctx):
    """List the available commands."""
    return [entry_point.name for entry_point in
            pkg_resources.iter_entry_points('chanjo.subcommands')]

  def get_command(self, ctx, name):
    """Load one of the available commands."""
    return pkg_resources.load_entry_point('chanjo', 'chanjo.subcommands', name)
```

Then when you define your CLI:

```python
@click.group(cls=EntryPointsCLI)
```